edx_toggles/toggles/state.py

```diff
@@ -0,0 +1,265 @@
+"""
+Toggle state report API.
+"""
+from collections import OrderedDict
+
+from django.conf import settings
+from edx_django_utils.monitoring import get_code_owner_from_module
+from waffle.models import Flag, Switch
+
+from . import SettingDictToggle, SettingToggle, WaffleFlag, WaffleSwitch
+
+
+class ToggleStateReport:
+    """
+    Convenient class to collect toggle objects across a codebase and expose them via a REST API.
+
+    Many methods are overridable to expose custom toggle objects in different IDAs.
+
+    Use as follows:
+
+        report = ToggleStateReport().as_dict()
```

Comment:
Sorry if I missed it, but where are docs for recommended view code? I think we have a how to for adding toggle reporting for other IDAs, and maybe something belongs there? Not a blocker, but better to not leave all doc updates till later.

Reply:
It's difficult for me to recommend the best view implementation, because I'm not sure how permissions should be handled. If it were up to me, I would not resort to class-based views and would simply write:

```
def get_toggle_state_report(request):
    return JsonResponse(ToggleStateReport().as_dict())
```

Reply:
It might be fine to start by pointing to the edx-platform code simply as an example.

If you want to use a function rather than a class, you could update that in edx-platform.  See https://github.com/edx/edx-platform/pull/26325#discussion_r581341136 for details.